stacks-common/src/types/net.rs

```diff
@@ -218,6 +218,21 @@ impl PeerAddress {
         }
     }
 
+    /// Is this a local loopback address?
+    pub fn is_loopback(&self) -> bool {
+        if self.is_ipv4() {
+            // 127.0.0.0/8
+            self.0[12] == 127
+        } else {
+            // ::1/128
+            *self
+                == PeerAddress([
+                    0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+                    0x00, 0x00, 0x01,
+                ])
+        }
```

Comment:
I realize its less efficient to do:

```rust
pub fn is_loopback(&self) -> bool {
     self.to_socketaddr(0).ip().is_loopback()
}
```

But I think its preferable to not implement our own IPv4 and IPv6 spec. In general, I think `PeerAddress` should just be replaced by `std::net::IpAddr`

Reply:
I agree, I was just following the patterns. Is there some reason we're not using Rust's standard lib here?

Reply:
@kantai Do you want this change now, or as part of a larger refactor to remove `PeerAddress`? Kinda looks weird having one function convert to `IpAddr` while similar ones don't

Reply:
Yes, I'd prefer the change now -- no reason to add our own implementation of a loopback check.